frontend/src/routes/Governance/components/ViewDiffApiCall.tsx

```diff
@@ -0,0 +1,102 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { useEffect, useState } from 'react'
+import { fireManagedClusterView } from '../../../resources'
+import { resultsTableData } from '../policies/policy-details/PolicyDetailsResults'
+import { useTranslation } from '../../../lib/acm-i18next'
+import { Alert, Button, Divider, Modal, ModalVariant, Skeleton, Title } from '@patternfly/react-core'
+import { Grid } from '@mui/material'
+import { TemplateDetailTitle } from './TemplateDetailTitle'
+import { CodeBlock } from './CodeBlock'
+
+export function ViewDiffApiCall({ item }: Readonly<{ item: resultsTableData }>) {
+  const { templateName, apiVersion, kind, cluster } = item
+  const [templateErr, setTemplateErr] = useState<string | undefined>()
+  const [isFetching, setIsFetching] = useState<boolean>(false)
+  const [relatedObjs, setRelatedObjs] = useState<any[]>()
+  const [isModalOpen, setIsModalOpen] = useState(false)
+
+  const { t } = useTranslation()
+  const handleModalToggle = () => {
+    setIsModalOpen((prevIsModalOpen) => !prevIsModalOpen)
+  }
+  useEffect(() => {
+    if (isModalOpen) {
+      setIsFetching(true)
+      setRelatedObjs(undefined)
+      setTemplateErr(undefined)
+
+      fireManagedClusterView(cluster, kind, apiVersion, templateName, cluster)
+        .then((viewResponse) => {
+          if (viewResponse?.message) {
+            setTemplateErr(viewResponse.message)
+          } else {
+            setRelatedObjs(viewResponse.result.status?.relatedObjects)
+          }
+          setIsFetching(false)
+        })
+        .catch((err) => {
+          console.error('Error getting resource: ', err)
+          setTemplateErr(err)
+          setIsFetching(false)
+        })
+    }
+  }, [isModalOpen, kind, apiVersion, templateName, cluster, t])
+
+  return (
+    <>
+      <span> {'-'} </span>
+      <div style={{ display: 'inline-block' }}>
+        <Button variant="link" isInline onClick={handleModalToggle} width={50}>
+          {t('View Diff')}
+        </Button>
+      </div>
+
+      <Modal
+        bodyAriaLabel="policy difference modal"
+        tabIndex={0}
+        header={
+          <>
+            <Title headingLevel="h1">
+              <TemplateDetailTitle policyKind={item.kind} templateName={item.templateName} compliant="NonCompliant" />
+            </Title>
+            <Divider style={{ marginTop: '10px' }} />
+            <Divider />
+          </>
+        }
+        variant={ModalVariant.large}
+        isOpen={isModalOpen}
+        onClose={handleModalToggle}
+        actions={[
+          <Button key="Cancel" variant="primary" onClick={handleModalToggle}>
+            {t('Close')}
+          </Button>,
+        ]}
+      >
+        {isFetching && (
+          <div style={{ height: '300px' }}>
+            <Skeleton height="100%" screenreaderText={t('Loading Template diff')} />
+          </div>
```

Comment:
This makes it look similar to when the diff is present.
```suggestion
<div style={{ height: '300px', marginTop: '30px', overflowY: 'hidden' }}>
            <Skeleton
              style={{
                borderRadius: '6px',
                backgroundColor: '#f6f8fa',
                border: '1px solid #d0d7de',
              }}
              height="100%"
              screenreaderText={t('Loading Template diff')}
            />
          </div>
```